File: pinyougou_parent/web_manager/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
    //读取列表数据绑定到表单中
    this.findAll=function(){
        return $http.get('../specification/findAll.do');
    }

    //分页查询  带条件
    this.search=function(page,rows,searchEntity){
        return $http.post('../specification/search.do?page='+page+'&rows='+rows,searchEntity);
    }

    //修改规格信息
    this.update=function (entity) {
        return $http.post('../specification/update.do',entity);
    }

    //增加规格信息
    this.add=function (entity) {
        return $http.post('../specification/add.do',entity);
    }

    //查询实体  修改的时候要对数据回显  传入id值根据id查询数据
    this.findOne=function (id) {
        return $http.get('../specification/findOne.do?id='+id);
    }

    //批量删除
    this.dele=function (ids) {
        return $http.post('../specification/delete.do?ids='+ids);
    }

    this.selectOptionList = function(){
        return $http.get("../specification/selectOptionList.do");
    }
})

File: pinyougou_parent/interface/src/main/java/com/xyxy/core/service/SpecificationService.java
package com.xyxy.core.service;

import com.xyxy.core.pojo.entity.PageResult;
import com.xyxy.core.pojo.entity.SpecEntity;
import com.xyxy.core.pojo.specification.Specification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    //查询所有
    public List<Specification> findAll();
    //分页查询 带条件
    public PageResult findPage(Specification specification,Integer pageNum , Integer rows);
    //规格添加
    public void add(SpecEntity specEntity);
    //查询实体
    public SpecEntity findOne(Long id);
    //规格修改
    public void update(SpecEntity specEntity);
    //删除规格
    public void dele(Long[] ids);
    //模板管理添加其模板时回显规格
    public List<Map> selectOptionList();
}
